docs: Add AI / LLM policy (#3300)
Pycord-Development/pycord: CONTRIBUTING.md

Reviewer: ```suggestion
By submitting a pull request, you agree that; 1) You hold the copyright on all submitted
code inside said pull request; 2) You agree to transfer all rights to the owner of this
repository; 3) If you are found to be in fault with any of the above, we shall not
be held responsible in any way after the pull request has been merged, and ; 4) You understand 
how your submitted code works and are able to explain its purpose, behavior, and impact when requested.
```

Reviewer: this block fucked up

Reviewer: addition of 4: You understand how your submitted code works and are able to explain its purpose, behavior, and impact when requested

Author: Should that really be under Licensing though then ?

Reviewer: either there or the section beyond. im not sure. 
tempted to add it to the last added line, but it mentions "as outlined in licensing", so it felt weird

Reviewer 2: This wasn't changed by this PR, but to bring it in line with the formatting that WAS added in this PR, I'd suggest making it a regular bullet point list like below

another. If that is the case, it is **required** that a comment is left explaining why
you are deciding to ignore type checking warnings.

## Licensing

By submitting a pull request, you agree that:

1. You hold the copyright on all submitted code inside said pull request;
2. You agree to transfer all rights to the owner of this repository, and;
3. If you are found to be in fault with any of the above, we shall not be held

>    responsible in any way after the pull request has been merged.

Reviewer 2: ```suggestion
By submitting a pull request, you agree to and accept the following:

1. You hold the copyright on all submitted code inside said pull request
2. You transfer all rights to the owner of this repository
3. If you are found to be in fault with any of the above, we shall not be held
   responsible in any way after the pull request has been merged.
```

>    responsible in any way after the pull request has been merged.

Reviewer 3: Items 1 and 2 have no trailing period, but item 3 does. This is inconsistent.
```suggestion
1. You hold the copyright on all submitted code inside said pull request.
2. You transfer all rights to the owner of this repository.
3. If you are found to be in fault with any of the above, we shall not be held
   responsible in any way after the pull request has been merged.
```

We don't limit nor deny your pulls when you're using another style although, please make
sure it is appropriate and makes sense in this library.

### AI Usage Policy

The use of LLMs or other AI tools for any kind of partial or complete interaction or

> contribution (included but not limited to writing code, documentation or discussing with

Reviewer 4: ```suggestion
contribution (including but not limited to the following: writing code/documentation or discussing with
```

maintainers) must be disclosed at all times. The disclosure must include what tools what

> used and within which scope.

Reviewer 4: ```suggestion
used and to what scopes they were used.
```

The use of fully autonomous LLMs or other AI tools to perform the action of opening

> issues or pull requests is forbidden.

Reviewer 3: ```suggestion
The use of fully autonomous LLMs or other AI tools to open issues or pull requests is forbidden.
```

Failure to disclose your AI use or to follow this policy may result in a temporary or
permanent ban from our organization.

The use of LLMs or AI tools by new contributors is highly discouraged, and may result to

> direct closure of your issue or pull request.

Reviewer 3: ```suggestion
The use of LLMs or AI tools by new contributors is highly discouraged, and may result in
direct closure of your issue or pull request.
```

> Furthermore, as outlined earlier in § "Licensing", by submitting a pull request, you

Reviewer 4: ```suggestion
Furthermore, as outlined earlier in § "Licensing," by submitting a pull request, you
```

We don't limit nor deny your pulls when you're using another style although, please make
sure it is appropriate and makes sense in this library.

## AI Usage Policy

The use of LLMs or other AI tools for any kind of partial or complete interaction or
contribution (including but not limited to the following: writing code/documentation or
discussing with maintainers) must be disclosed at all times. The disclosure must include

> what tools were used and to what scopes they were used.

Reviewer 3: The "were used" is repeated twice for no reason. Tautology.
```suggestion
what tools were used and to what scope.
```

Author: @JustaSqu1d

Reviewer 4: "what scope" is now a fragment. What about
> The disclosure must specify the tools used and the scope of their application.

Reviewer 3: Sure

If you are not a native English speaker, using LLMs or other AI tools for translation is
discouraged. It can distort your intended meaning and make a human-written contribution

> look AI-generated. Please use a non-LLM-based translation tool instead.

Reviewer: ```suggestion
look AI-generated. Please use a non-LLM-based translation tool like https://deepl.com instead.
```

The use of fully autonomous LLMs or other AI tools to open issues or pull requests is

> forbidden.

Reviewer: ```suggestion
strictly forbidden.
```

We don't limit nor deny your pulls when you're using another style although, please make
sure it is appropriate and makes sense in this library.

## AI Usage Policy

The use of LLMs or other AI tools for any kind of partial or complete interaction or
contribution (including but not limited to the following: writing code/documentation or

> discussing with maintainers) must be disclosed at all times. The disclosure must include

Reviewer 3: Weird to use both / and "or" for enumeration.
```suggestion
contribution (including but not limited to the following: writing code, documentation, or
discussing with maintainers) must be disclosed at all times. The disclosure must include
```

Author: This was changed from , to / by squ1d.

@JustaSqu1d do you think should I change it back ?

Reviewer 4: If we want to keep it, then there should be a semicolon before "or" and change the comma to an "or." Rephrasing this clause is probably a better alternative

> what tools what used and to what scopes they were used.

Reviewer 3: ```suggestion
what tools were used and to what scope.
```

Author: Idem @JustaSqu1d , regarding the second part

> what tools were used and to what scopes they were used.

Reviewer 3: ```suggestion
discussing with maintainers) must be disclosed at all times. The disclosure must specify 
the tools used and the scope of their application.
```

Using LLMs or other AI tools for translation, for example, if you are not a native

> English speaker, is discouraged. It can distort your intended meaning and make a

Reviewer 3: ```suggestion
If you are not a native English speaker, using LLMs or other AI tools for translation is
discouraged. It can distort your intended meaning and make a
```

look AI-generated. Please use a non-LLM-based translation tool like https://deepl.com
instead.

The use of fully autonomous LLMs or other AI tools to open issues or pull requests is
strictly forbidden.

Failure to follow this policy may result in a temporary or permanent ban from our
organization.

The use of LLMs or AI tools by new contributors is highly discouraged, and may result in
the direct closure of your issue or pull request.

Furthermore, as outlined earlier in §"Licensing," by submitting a pull request, you

> agree that you hold the copyright on your contribution.

Reviewer: ```suggestion
Furthermore, as outlined earlier in §"Licensing," by submitting a pull request, you
agree that you hold the copyright on your contribution; and that you understand how your submitted code works and are able to explain its purpose, behavior, and impact when requested.
```

alternative

We don't limit nor deny your pulls when you're using another style although, please make
sure it is appropriate and makes sense in this library.

## AI Usage Policy

The use of LLMs or other AI tools for any kind of partial or complete interaction or

> contribution (including but not limited to the following: writing code/documentation or

Reviewer 4: ```suggestion
contribution (including but not limited to the following: code, documentation, or
```

Author: So we end up doing what Vmphase did 😭 alr

Reviewer 4: no, we removed the verb "writing" so the parallelism is kept.

> discussing with maintainers) must be disclosed at all times. The disclosure must specify

Reviewer 4: ```suggestion
discussions with maintainers) must be disclosed at all times. The disclosure must specify
```

> specify the tools used and the scope of their application.

Reviewer 2: This sentence is a bit nested and long, I'd prefer a bullet-point'd list of things that must be disclosed. Also, note where this must happen (e.g. PR template if that was updated already)

Author: I agree but I don't have energy rn. Feel free to propose something or I will see at some point later

Reviewer 5: Might need to reformat the line breaks
```suggestion
The use of LLMs or other AI tools must be disclosed at all times.
Disclosure is required for any interaction or contribution which includes but is not limited to:
- Modifying Code
- Modifying Documentation
- Discussion with maintainers
- Issue bodies and comments
- Pull Request bodies and comments
The disclosure must specify the tools used and the scope of their application.
The primary location of disclosure should be in the body of Issues and Pull Requests.
```

If you are not a native English speaker, using LLMs or other AI tools for translation is
discouraged. It can distort your intended meaning and make a human-written contribution

> look AI-generated. Please use a non-LLM-based translation tool like https://deepl.com

Reviewer 4: ```suggestion
look AI-generated. Please use a non-LLM-based translation tool like [DeepL](https://deepl.com)
```

[DeepL](https://deepl.com) instead.

> The use of fully autonomous LLMs or other AI tools to open issues or pull requests is

Reviewer 2: This is a bit vague and could confuse people. You probably (even I'm not sure) mean 100% AI generated PRs from an Agent, in which case, just name that

Author: No I mean what I said. Opening a pr with a fully autonomous tool even if you wrote it yourself

Author: Also the entire point is to not use restrictive naming because otherwise people are gonna find a way to complain

Reviewer 2: Okay then I'm confused what you exactly mean by "fully autonomous tool"?

Reviewer: people should write open and write the pr desc themself

strictly forbidden.

Failure to follow this policy may result in a temporary or permanent ban from our
organization.

The use of LLMs or AI tools by new contributors is highly discouraged, and may result in

> the direct closure of your issue or pull request.

Reviewer 2: I'm not sure but this might be a bit harsh. A newbie writing his own thing and letting AI read over it and find 1-2 mistakes lowers the barrier to entry while not being meaningfully "AI"

Author: No not harsh sorry. If someone unknown just comes here to contribute something and does it with AI they deserve it being closed. We want to be welcoming for newbies as long as they wanna learn stuff.

Furthermore, as outlined earlier in §"Licensing," by submitting a pull request, you
agree that you hold the copyright on your contribution; and that you understand how your
submitted code works and are able to explain its purpose, behavior, and impact when

> requested.

Reviewer 3: ```suggestion
agree that you hold the copyright on your contribution, and that you understand how your
contribution works and are able to explain its purpose, behavior, and impact when
requested.
```

Author: why removing "your"

Reviewer: *hiss* out of my section

Reviewer 3: By mistake. Corrected.

Reviewer: contribution does not make any sense. we talk code here.

Author: Also semicolon makes more sense imo to separate from the licensing part of things

Reviewer 3: What if the PR is not code-related, but instead a docs change (f.e.)?  
The AI policy applies to that as well.

Reviewer: find a different sentence then. most contributions are code tho

We don't limit nor deny your pulls when you're using another style although, please make
sure it is appropriate and makes sense in this library.

## AI Usage Policy

The use of LLMs or other AI tools must be disclosed at all times. Disclosure is required
for any interaction or contribution which includes but is not limited to:

- Modifying Code
- Modifying Documentation
- Discussion with maintainers
- Issue bodies and comments

> - Pull Request bodies and comments The disclosure must specify the tools used and the

Reviewer 2: Something went wrong here with the newlines, please fix

  scope of their application. The primary location of disclosure should be in the body

>   of Issues and Pull Requests.

Reviewer 2: This entire section uses title case for some reason, especially for words like "Pull Request" and "Issues" which are lowercased later on.

We don't limit nor deny your pulls when you're using another style although, please make
sure it is appropriate and makes sense in this library.

## AI Usage Policy

The use of LLMs or other AI tools must be disclosed at all times. Disclosure is required

> for any interaction or contribution which includes but is not limited to:

Reviewer 4: ```suggestion
for any interaction or contribution which includes but is not limited to the following:
```

> for any interaction or contribution including but not limited to the following:

Reviewer 4: ```suggestion
for any interaction or contribution including, but not limited to, the following:
```

> for any interaction or contribution which includes but is not limited to the following:

Reviewer 3: ```suggestion
for any interaction or contribution including but not limited to the following:
```

- Modifying code
- Modifying documentation
- Discussion with maintainers
- Issue bodies and comments

> - Pull request bodies and comments

Reviewer 2: ```suggestion
- Pull Request bodies and comments
```

Author: Litterally We don't use it in the rest I changed it on purpose toothy

The disclosure must specify the tools used and the scope of their application. The

> primary location of disclosure should be in the body of Issues and Pull Requests.

Reviewer 3: Inconsistent capitalization: "Issues and Pull Requests" is capitalized here, but "issues and/or pull requests" is lowercase elsewhere. Furthermore, GitHub itself renders it as "Pull requests" (lowercase second word).
```suggestion
The disclosure must specify the tools used and the scope of their application. The
primary location of disclosure should be in the body of issues and pull requests.
```

If you are not a native English speaker, using LLMs or other AI tools for translation is
discouraged. It can distort your intended meaning and make a human-written contribution

> look AI-generated. Please use a non-LLM-based translation tool like

Reviewer 4: ```suggestion
look AI-generated. Please use a non-LLM-based translation tool such as
```

[DeepL](https://deepl.com) instead.

> The use of fully autonomous LLMs or other AI tools to open issues or pull requests is

Reviewer 2: ```suggestion
The use of fully autonomous LLMs or other AI tools to open Issues or Pull Requests is
```

strictly forbidden.

Failure to follow this policy may result in a temporary or permanent ban from our
organization.

The use of LLMs or AI tools by new contributors is highly discouraged, and will likely
result in the direct closure of your issue or pull request.

Furthermore, as outlined earlier in §"Licensing," by submitting a pull request, you

> agree that you hold the copyright on your contribution; and that you understand how your

Reviewer 3: A semicolon should join two independent clauses. "and that you understand..." is not independent, but rather a second object of "agree that X and that Y," so this needs a comma, not a semicolon.
```suggestion
agree that you hold the copyright on your contribution, and that you understand how your
```
